### DISCUSSION TITLE
ANT+ FE-C to FTMS Dongle/ESP32 board, etc?

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [doudar/SmartSpin2k#608] [Q&A] [answered]
Sorry if this is the wrong place to ask this.

I'm coming back to this after a few years of cobbling other solutions together...here's my problem I'm trying to solve.  I use Garmin for my fitness tracker/watch.  It has a program that designs a cycling program based on your goals and takes into consideration your current workout load etc to build the weekly rides.  Problem is that it only communicates to trainers on the ANT+ FE-C protocol.  I've seen some dongles mentioned on the internet like NPE's WYUR that supposedly will take the Garmin Watch's FE-C and re-broadcast it as a FTMS but they seemingly don't sell them anymore (same goes for really any of the other few options).

I've done some simple things with arduino/esp32 in the past so was wondering if this is something that a non-programmer could build rather easily or does it require inspecting the bluetooth transmissions, finding the ID's and a lot of coding etc to make it work.

What I'm currently doing is taking the Garmin suggested workout, manually building it in intervals.icu, then exporting the .ZWO file that it creates and manually importing it into https://github.com/dvmarinoff/Auuki which connects to SS2K to control the resistance.  I'd much rather just go straight from the Garmin to the SS2K via a ANT+ FE-C to Bluetooth bridge, either one I can buy or make.

I appreciate any feedback/thoughts!

**Comment by SchneiderMK** [comment, marked as answer]
It is not an easy task. You need extra Hardware. ESP32 can not read ANT+ without an extension board and you must understand both protocols FE-C and FTMS to create the mapping between them.
